cmake/build_helpers.cmake

```diff
@@ -18,11 +18,13 @@ function(ginkgo_default_includes name)
 endfunction()
 
 function(ginkgo_compile_features name)
-    target_compile_features("${name}" PUBLIC cxx_std_14)
-    # we set these properties regardless of the enabled backends,
-    # because unknown properties are ignored
-    set_target_properties("${name}" PROPERTIES HIP_STANDARD 14)
-    set_target_properties("${name}" PROPERTIES CUDA_STANDARD 14)
+    target_compile_features("${name}" PUBLIC cxx_std_17)
+    if (GINKG_BUILD_CUDA)
+        target_compile_features("${name}" PUBLIC cuda_std_17)
+    endif()
+    if (GINKG_BUILD_HIP)
+        target_compile_features("${name}" PUBLIC hip_std_17)
+    endif()
```

Comment:
Properties are not transitive, while compile features are. `hip_std_17` already appears in the test code for CMake 3.21.1, so it was just not documented then.

Reply:
If it wasn't already supported in 3.21, the test would fail. I verified the existence and functionality of the feature manually with 3.21.1. The documentation for it was added later, see [CMAKE_HIP_KNOWN_FEATURES](https://cmake.org/cmake/help/latest/prop_gbl/CMAKE_HIP_KNOWN_FEATURES.html#prop_gbl:CMAKE_HIP_KNOWN_FEATURES). CMake throws an error on unknown features, so we would notice if it was missing.